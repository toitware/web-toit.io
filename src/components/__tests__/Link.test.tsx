/**
 * @jest-environment jsdom
 */
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import { StableClassNames } from "@toitware/testing-utils";
import React from "react";
import Link from "../link";

describe("Block", () => {
  it("uses GatsbyLink if 'to'", () => {
    const result = render(
      <StableClassNames>
        <Link to="/home">foobar</Link>
      </StableClassNames>
    );

    expect(result.container.firstChild).toMatchSnapshot();
  });
  it("uses a element if href is provided", () => {
    const result = render(
      <StableClassNames>
        <Link href="https://foobar">foobar</Link>
      </StableClassNames>
    );

    expect(result.container.firstChild).toMatchSnapshot();
  });
});
