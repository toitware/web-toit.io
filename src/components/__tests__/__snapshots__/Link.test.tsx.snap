// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Block uses GatsbyLink if 'to' 1`] = `
<a
  href="/home"
>
  foobar
</a>
`;

exports[`Block uses a element if href is provided 1`] = `
<a
  href="https://foobar"
  rel="noreferrer"
  target="_blank"
>
  foobar
</a>
`;
